### DISCUSSION TITLE
FullLifecycleObserver dependency error

### DISCUSSION THREAD

**Discussion by gulcesenyuz** [skydoves/Balloon#498] [Q&A]
Cannot access 'androidx.lifecycle.FullLifecycleObserver' which is a supertype of 'com.skydoves.balloon.Balloon'. Check your module classpath for missing or conflicting dependencies

Can you help me to solve this

**Comment by faraz152** [comment]
this happens when your project forces a newer `lifecycle-common` version that removed the internal `FullLifecycleObserver` class. the fix is to update Balloon to the latest version — newer releases use `DefaultLifecycleObserver` instead which is the public API in lifecycle 2.4+.

if upgrading Balloon is not an option, you can pin lifecycle-common to 2.3.1 in your gradle but that will hold back other lifecycle deps too.
